teuthology/test/task/__init__.py

```diff
@@ -0,0 +1,132 @@
+from mock import patch
+from pytest import raises
+
+from teuthology.config import FakeNamespace
+from teuthology.orchestra.cluster import Cluster
+from teuthology.orchestra.remote import Remote
+from teuthology.task import Task
+
+
+class TestTask(object):
+    def setup(self):
+        self.ctx = FakeNamespace()
+        self.ctx.config = dict()
+
+    def test_overrides(self):
+        self.ctx.config['overrides'] = dict(
+            task=dict(
+                key_1='overridden',
+            ),
+        )
+        task_config = dict(
+            key_1='default',
+            key_2='default',
+        )
+        with Task(self.ctx, task_config) as task:
+            assert task.config['key_1'] == 'overridden'
+            assert task.config['key_2'] == 'default'
+
+    def test_hosts_no_results(self):
+        self.ctx.cluster = Cluster()
+        self.ctx.cluster.add(Remote('remote1'), ['role1'])
+        task_config = dict(
+            hosts=['role2'],
+        )
+        with raises(RuntimeError):
+            with Task(self.ctx, task_config) as task:
+                task_hosts = task.hosts
+                assert len(task_hosts) == 0
```

Comment:
You're right; I'd put those two lines in before I wrapped the block with ``with raises``.